app/src/main/java/org/fossasia/susi/ai/settings/ChatSettingsFragment.kt

```diff
@@ -77,19 +81,40 @@ class ChatSettingsFragment : PreferenceFragmentCompat() {
 
         theme?.setOnPreferenceChangeListener({ preference, newValue ->
             preference.summary = newValue.toString()
-            settingsPresenter?.setTheme(newValue.toString())
+            settingsPresenter.setTheme(newValue.toString())
+            settingsPresenter.sendSetting(Constant.THEME, newValue.toString())
             activity.recreate()
             true
         })
 
-        micSettings?.isEnabled = settingsPresenter?.enableMic() as Boolean
+        micSettings?.isEnabled = settingsPresenter.enableMic()
+
+        hotwordSettings?.isEnabled = settingsPresenter.enableHotword()
 
-        hotwordSettings?.isEnabled = settingsPresenter?.enableHotword() as Boolean
+        micSettings?.setOnPreferenceClickListener {
+            settingsPresenter.sendSetting(Constant.MIC_INPUT, (PrefManager.getBoolean(Constant.MIC_INPUT, false)).toString())
+            true
+        }
+
+        enterSend?.setOnPreferenceChangeListener { _, newValue ->
+            settingsPresenter.sendSetting(Constant.ENTER_SEND, newValue.toString())
+            true
+        }
+
+        speechAlways?.setOnPreferenceChangeListener { _, newValue ->
+            settingsPresenter.sendSetting(Constant.SPEECH_ALWAYS, newValue.toString())
+            true
+        }
+
+        speechOutput?.setOnPreferenceChangeListener { _, newValue ->
+            settingsPresenter.sendSetting(Constant.SPEECH_OUTPUT, newValue.toString())
+            true
+        }
```

Comment:
There was listener for hotword detection too before shifting settings to MVP. Make sure everything which was there earlier, exists now too.